conn_process.go

```diff
@@ -116,6 +116,11 @@ func (c *connect) process(ctx context.Context, on *onProcess) error {
 	select {
 	case <-ctx.Done():
 		c.cancel()
+		// Wait for goroutine to finish before returning
+		select {
```

Comment:
My bit of concern with this approach is, if you look at the background goroutine, it doesn't use/watch for passed in `ctx` for cancellation. It does give it to upstream `handle` method.

the problem is, the whole point of canceling this method is to stop doing `handle` and return immediately. 

With this change, we are indirectly may end-up waiting for `handle` to "finish". 


I would handle the situation ideally like following.
1. make the goroutine listen with two `select` case. a. handle to finish b. context to be done.
2. Based on which ever comes first (a) or (b), we exit the go-routine.

Does it make sense?